engines/gob/inter_v7.cpp

```diff
@@ -960,25 +1107,715 @@ void Inter_v7::o7_loadIFFPalette() {
 	_vm->_video->setFullPalette(_vm->_global->_pPaletteDesc);
 }
 
-void Inter_v7::o7_opendBase() {
+void Inter_v7::o7_openDatabase() {
+	Common::String databaseName = _vm->_game->_script->evalString();
+	_vm->_game->_script->readValExpr(); // unknown, some kind of "open mode"
+
+	_databases.setVal(databaseName, Database());
+}
+
+void Inter_v7::o7_openDatabaseTable() {
+	Common::String databaseName = _vm->_game->_script->evalString();
+	Common::String tableName = _vm->_game->_script->evalString();
+	Common::String dbFile = getFile(_vm->_game->_script->evalString());
+
+	dbFile += ".DBF";
+
+	if (!_databases.contains(databaseName)) {
+		warning("o7_openDatabaseTable(): No such database \"%s\"", databaseName.c_str());
+		return;
+	}
+
+	Database &database = _databases.getVal(databaseName);
+
+	database.openTable(tableName, Common::Path(dbFile));
+}
+
+void Inter_v7::o7_closeDatabaseTable() {
+	Common::String databaseName = _vm->_game->_script->evalString();
+	Common::String tableName = _vm->_game->_script->evalString();
+
+	if (!_databases.contains(databaseName)) {
+		warning("o7_closeDatabaseTable(): No such database \"%s\"", databaseName.c_str());
+		return;
+	}
+
+	Database &database = _databases.getVal(databaseName);
+	database.closeTable(tableName);
+}
+
+void Inter_v7::o7_draw0xAE() {
+	_vm->_game->_script->readValExpr();
+	_vm->_game->_script->readValExpr();
+	warning("STUB: o7_draw0xAE (Adibou/Musique)");
+}
+
+void Inter_v7::o7_openDatabaseIndex() {
+	Common::String databaseName = _vm->_game->_script->evalString();
+	Common::String tableName = _vm->_game->_script->evalString();
+	Common::String indexName = _vm->_game->_script->evalString();
+
+	if (!_databases.contains(databaseName)) {
+		warning("o7_openDatabaseIndex(): No such database \"%s\"", databaseName.c_str());
+		return;
+	}
+
+	Database &database = _databases.getVal(databaseName);
+	dBase *db = database.getTable(tableName);
```

Comment:
Well, now I see that we may make an exception here, since `dBase` is the format name.